src/game_records.c

```diff
@@ -0,0 +1,416 @@
+   [RECORD_UNK_143] = FALSE,
+   [RECORD_UNK_144] = FALSE,
+   [RECORD_UNK_145] = FALSE,
+   [RECORD_UNK_146] = FALSE,
+   [RECORD_UNK_147] = FALSE,
+};
+
+static u32 GetRecordLimit(int id)
+{
+    if (id < NUM_U32_RECORDS) {
+        if (sUsesHighLimit[id]) {
+            return HIGH_LIMIT_U32;
+        } else {
+            return LOW_LIMIT_U32;
+        }
+    } else if (id < MAX_RECORDS) {
+        if (sUsesHighLimit[id]) {
+            return HIGH_LIMIT_U16;
+        } else {
+            return LOW_LIMIT_U16;
+        }
+    }
+
+    GF_ASSERT(0);
+    return 0;
+}
+
+static const u16 sTrainerScoreIncrements[MAX_SCORE_RECORDS] = {
+    [SCORE_RECORD_UNK_00] = 1,
+    [SCORE_RECORD_UNK_01] = 1,
+    [SCORE_RECORD_UNK_02] = 1,
+    [SCORE_RECORD_UNK_03] = 1,
+    [SCORE_RECORD_UNK_04] = 1,
+    [SCORE_RECORD_UNK_05] = 1,
+    [SCORE_RECORD_UNK_06] = 2,
+    [SCORE_RECORD_UNK_07] = 2,
+    [SCORE_RECORD_UNK_08] = 2,
+    [SCORE_RECORD_UNK_09] = 2,
+    [SCORE_RECORD_UNK_10] = 3,
+    [SCORE_RECORD_UNK_11] = 3,
+    [SCORE_RECORD_UNK_12] = 3,
+    [SCORE_RECORD_UNK_13] = 7,
+    [SCORE_RECORD_UNK_14] = 7,
+    [SCORE_RECORD_UNK_15] = 7,
+    [SCORE_RECORD_UNK_16] = 10,
+    [SCORE_RECORD_UNK_17] = 10,
+    [SCORE_RECORD_UNK_18] = 11,
+    [SCORE_RECORD_UNK_19] = 11,
+    [SCORE_RECORD_UNK_20] = 11,
+    [SCORE_RECORD_UNK_21] = 11,
+    [SCORE_RECORD_CAUGHT_SPECIES] = 20,
+    [SCORE_RECORD_UNK_23] = 30,
+    [SCORE_RECORD_UNK_24] = 35,
+    [SCORE_RECORD_UNK_25] = 40,
+    [SCORE_RECORD_UNK_26] = 500,
+    [SCORE_RECORD_UNK_27] = 10000,
+    [SCORE_RECORD_UNK_28] = 30,
+    [SCORE_RECORD_UNK_29] = 30,
+    [SCORE_RECORD_UNK_30] = 2,
+    [SCORE_RECORD_UNK_31] = 5,
+    [SCORE_RECORD_UNK_32] = 1,
+    [SCORE_RECORD_UNK_33] = 1,
+    [SCORE_RECORD_UNK_34] = 5,
+    [SCORE_RECORD_UNK_35] = 3,
+    [SCORE_RECORD_UNK_36] = 1,
+    [SCORE_RECORD_UNK_37] = 1,
+    [SCORE_RECORD_UNK_38] = 7,
+    [SCORE_RECORD_UNK_39] = 7,
+    [SCORE_RECORD_UNK_40] = 7,
+    [SCORE_RECORD_UNK_41] = 7,
+    [SCORE_RECORD_UNK_42] = 1000,
+    [SCORE_RECORD_UNK_43] = 11,
+    [SCORE_RECORD_UNK_44] = 20,
+    [SCORE_RECORD_UNK_45] = 10,
+    [SCORE_RECORD_UNK_46] = 15,
+    [SCORE_RECORD_UNK_47] = 11,
+    [SCORE_RECORD_UNK_48] = 11,
+    [SCORE_RECORD_UNK_49] = 10,
+    [SCORE_RECORD_UNK_50] = 10,
+};
+
+static int GetTrainerScoreIncrement(int records)
+{
+    return sTrainerScoreIncrements[records];
+}
+
+static inline u32 SetRecordValueWithLimit(GameRecords *records, int id, u32 val, u32 limit)
+{
+    return val < limit
+        ? SetRecordValue(records, id, val)
+        : SetRecordValue(records, id, limit);
+}
+
+u32 GameRecords_SetRecordValue(GameRecords *records, int id, u32 val)
+{
+    u32 limit = GetRecordLimit(id);
+
+    DecodeGameRecords(records, id);
+    u32 new = SetRecordValueWithLimit(records, id, val, limit);
+    EncodeGameRecords(records, id);
+
+    return new;
+}
+
+u32 GameRecords_SetAndLimitRecordValue(GameRecords *records, int id, u32 val)
```

Comment:
I don't think this name is accurate? This function is different from `GameRecords_SetRecordValue` in that it sets the value only if the incoming value is greater than the current value, and both functions limit the value being set anyway.

Reply:
`GameRecords_SetGreaterRecordValue`?